--- client/jquery/public/client-app.js	2017-01-05 09:33:29.548365710 -0500
+++ client/webpack/client/app.js	2017-01-15 21:13:50.702013335 -0500
@@ -1,5 +1,15 @@
 
 /* globals feathersClient: 0, feathers: 0 */
+const moment = require('moment');
+const $ = require('jquery');
+const feathers = require('feathers-client');
+const io = require('socket.io-client');
+
+var serverUrl = 'http://localhost:3030';
+const socket = io(serverUrl);
+
+const feathersClient = feathers()
+  .configure(feathers.socketio(socket));
 
 // Routing
 
