--- 01/common/public/feathers-app.js	2017-01-13 10:16:46.208646185 -0500
+++ 02/app/public/feathers-app.js	2017-01-10 15:06:04.736345442 -0500
@@ -1,4 +1,11 @@
 
+feathersClient
+  .configure(feathers.hooks())
+  .configure(feathers.authentication({
+    storage: window.localStorage
+  }));
+
+let jackId;
 const users = feathersClient.service('/users');
 
 Promise.all([
@@ -13,7 +20,20 @@
     console.log('created Judy Doe item\n', results[2]);
     console.log('created Jack Doe item\n', results[3]);
     
-    return users.find()
-      .then(results => console.log('find all items\n', results));
+    jackId = results[3]._id;
+    
+    return feathersClient.authenticate({
+      type: 'local',
+      'email': 'jane.doe@gmail.com',
+      'password': '11111'
+    })
+      .then(() => console.log('\nAuthenticated successfully.\n '))
+      .catch(err => console.error('\nError authenticating:', err));
   })
+  .then(() => users.remove(jackId)
+    .then(results => console.log('deleted Jack Doe item\n', results))
+  )
+  .then(() => users.find()
+    .then(results => console.log('find all items\n', results))
+  )
   .catch(err => console.log(err));
