--- 01/hooks/1.js	2017-01-13 14:03:07.126341013 -0500
+++ 01/hooks/2.js	2017-01-13 14:02:23.933026428 -0500
@@ -1,5 +1,5 @@
 
-// Example - Use hooks with service
+// Example - Use soft deletes with a service
 
 const expressServerConfig = require('../common/expressServerConfig');
 const expressMiddleware = require('../common/expressMiddleware');
@@ -14,6 +14,10 @@
 const hooks = require('feathers-hooks');
 const commonHooks = require('feathers-hooks-common');
 
+
+
+
+
 const app = expressServerConfig()
   .configure(hooks())
   .configure(rest())
@@ -35,10 +39,12 @@
   const userService = app.service('users');
   
   const {
+    softDelete, when,
     setCreatedAt, setUpdatedAt, unless, remove /* , validateSchema */
   } = commonHooks;
   
   userService.before({
+    all: when(hook => hook.method !== 'find', softDelete()), // new
     create: [
       validateSchema(userSchema(), Ajv),
       authHooks.hashPassword(),
