--- 01/rest/1.js	2017-01-13 12:24:46.417687064 -0500
+++ 01/rest/2.js	2017-01-10 11:33:57.090334579 -0500
@@ -1,13 +1,14 @@
 
-// Example - Create REST API
+// Example - Create REST API, and serve static files
 
-const expressServerConfig = require('../common/expressServerConfig');
-const expressMiddleware = require('../common/expressMiddleware');
 const NeDB = require('nedb');
 const path = require('path');
 const service = require('feathers-nedb');
 const rest = require('feathers-rest');
 
+const expressServerConfig = require('../common/expressServerConfig');
+const expressMiddleware = require('../common/expressMiddleware');
+
 const app = expressServerConfig()
   .configure(rest())
   .configure(services)
